packages/client/window/src/windows/IFrame.ts

```diff
@@ -0,0 +1,54 @@
+import { EventMap } from "../EventEmitter";
+import { EventEmitterWithHandshakeOptions } from "../handshake";
+import { HandshakeParent } from "../handshake/Parent";
+
+export class IFrameWindow<IncomingEvents extends EventMap, OutgoingEvents extends EventMap> extends HandshakeParent<
+    IncomingEvents,
+    OutgoingEvents
+> {
+    iframe: HTMLIFrameElement;
+    targetOrigin: string;
+
+    private constructor(
+        iframe: HTMLIFrameElement,
+        targetOrigin: string,
+        options?: EventEmitterWithHandshakeOptions<IncomingEvents, OutgoingEvents>
+    ) {
+        const contentWindow = iframe.contentWindow;
+        if (!contentWindow) {
+            throw new Error("IFrame must have a contentWindow");
+        }
+        super(contentWindow, targetOrigin, options);
+
+        this.iframe = iframe;
+        this.targetOrigin = targetOrigin;
+    }
+
+    static async init<IncomingEvents extends EventMap, OutgoingEvents extends EventMap>(
+        url: string,
+        options?: EventEmitterWithHandshakeOptions<IncomingEvents, OutgoingEvents>
```

Comment:
From the example shared in the PR description and from my understanding, I see that the init method always should receive the `outgoingEvents` and the `incomingEvents`.

If so, should this be required?

Reply:
i think its fine, sometimes you dont want to send any events, it can be used to just open windows